### DISCUSSION TITLE
Dialog Plugin (Quasar V1): Closed dialogs leave some HTML at the end of the body

### DISCUSSION THREAD

**Discussion by drahkrub** [quasarframework/quasar#15455] [General - Components / Directives / etc]
i.e.
```
<div role="dialog" aria-modal="false" class="q-dialog fullscreen no-pointer-events q-dialog--seamless"><!----><!----></div>
```
In contrast ordinary `QDialog` components (opened via `v-model` not by the Dialog Plugin) seem to disappear completely from the HTML if closed.

Am I right or am I doing something wrong?

**Comment by johnlawsharrison** [comment]
Also noticing this and it's driving me crazy! From looking at the Vue inspector, it also seems like the plugin isn't cleaning up any Vue instances after a dialog is hidden.

**Comment by pdanpdan** [comment]
Do you have a reproduction codepens and step by step instructions to reproduce?

**Reply by johnlawsharrison** [reply]
I think I've identified the root cause, working on a CodePen but will finish after getting some sleep.

**Comment by yusufkandemir** [comment]
Make sure you have `@hide="onDialogHide"` on the `<q-dialog`. Also, ensure you have specified the emits, e.g. `defineEmits(useDialogPluginComponent.emits)`

**Reply by johnlawsharrison** [reply]
This is the cause, I neglected to bind `@hide="onDialogHide"` on my custom QDialog component.

**Reply by johnlawsharrison** [reply]
It might be worth adding a `REQUIRED` comment to the example for this in the docs: https://quasar.dev/quasar-plugins/dialog#sfc-with-script-setup-and-composition-api-variant. At first glance I was under the impression that setting up that binding was optional.

**Reply by johnlawsharrison** [reply]
Maybe it's possible that someone would want to keep the global dialog component around after hiding the QDialog itself (for re-use with a subsequent QDialog?), but that seems like non-default behavior.
